beacon_node/lighthouse_network/src/discovery/subnet_predicate.rs

```diff
@@ -1,15 +1,22 @@
 //! The subnet predicate used for searching for a particular subnet.
 use super::*;
 use crate::types::{EnrAttestationBitfield, EnrSyncCommitteeBitfield};
+use itertools::Itertools;
 use slog::trace;
 use std::ops::Deref;
+use types::{ChainSpec, DataColumnSubnetId};
 
 /// Returns the predicate for a given subnet.
-pub fn subnet_predicate<E>(subnets: Vec<Subnet>, log: &slog::Logger) -> impl Fn(&Enr) -> bool + Send
+pub fn subnet_predicate<E>(
+    subnets: Vec<Subnet>,
+    log: &slog::Logger,
+    spec: &ChainSpec,
+) -> impl Fn(&Enr) -> bool + Send
 where
     E: EthSpec,
 {
     let log_clone = log.clone();
+    let spec_clone = spec.clone();
```

Comment:
We can do this later, but we should probably avoid cloning the `ChainSpec` so frequently by `Arc`ing it

Reply:
To do it properly we should probably `Arc` it everywhere, but we could probably do it in a separate PR as it will be  quite a big change. I've `Arc`'d the spec only in `Discovery` and `NetworkService` to avoid the performance regression for now.